cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseVersion.java

```diff
@@ -243,13 +243,14 @@ static Version determineVersionFromVersionString(String versionString) throws Pa
     if (versionString.startsWith(HBASE_98_VERSION)) {
       return Version.HBASE_98;
     }
-    VersionNumber ver = VersionNumber.create(versionString);
     if (versionString.startsWith(HBASE_10_VERSION)) {
+      VersionNumber ver = VersionNumber.create(versionString);
```

Comment:
Also seems like `VersionNumber.create` is only used for CDH version detection. We can rename the method to make it more specific such as `createForCDH`

So `determineVersionFromVersionString` method would be

```
  if (versionString.contains("cdh") {
     return VersionNumber.create(versionString);
 }
// Do checks for hdp versions here..
```